be/src/exec/pipeline/sink/hive_table_sink_operator.cpp

```diff
@@ -199,18 +199,20 @@ void HiveTableSinkOperatorFactory::close(RuntimeState* state) {
 }
 
 void HiveTableSinkOperator::add_hive_commit_info(starrocks::parquet::AsyncFileWriter* writer, RuntimeState* state) {
-    THiveFileInfo hive_file_info;
-    hive_file_info.__set_file_name(path_util::base_name(writer->file_location()));
-    hive_file_info.__set_partition_path(writer->partition_location());
-    hive_file_info.__set_record_count(writer->metadata()->num_rows());
-    hive_file_info.__set_file_size_in_bytes(writer->file_size());
-
-    TSinkCommitInfo commit_info;
-    commit_info.__set_hive_file_info(hive_file_info);
-
-    // update runtime state
-    state->add_sink_commit_info(commit_info);
-    state->update_num_rows_load_sink(hive_file_info.record_count);
+    if (writer->metadata()) {
```

Comment:
1. please add comments to explain when metadata() will be null.
2. prefer `writer->metadata() != nullptr` that explicitly compares a pointer with the nullptr.

Reply:
I refactored a little bit, there would case in daily-test to cover this fix.